### PR TITLE
Upgrade to core-js v3

### PR DESCRIPTION
**Pull request by armandabric**
## What I did

Following the discussion in https://github.com/storybooks/storybook/pull/6589, this PR upgrade core-js to v3. It will help to upgrade the webpack CRA preset to match the CRA v3 release

I use the work done in https://github.com/storybooks/storybook/pull/6566 as guide

I think this should be considered as breaking change: it could lead to some behavior change on CRA storybook.

### CODE REVIEW

**Comment by Spy-Seth** [comment]
I've resolve the error that where directly trigger by some storybook code. But now i'm stuck on this kind of error:

```
ERROR in /Users/armand/workspace/storybook/node_modules/babel-runtime/core-js/array/from.js
Module not found: Error: Can't resolve 'core-js/library/fn/array/from' in '/Users/armand/workspace/storybook/node_modules/babel-runtime/core-js/array'
 @ /Users/armand/workspace/storybook/node_modules/babel-runtime/core-js/array/from.js 1:30-70
 @ /Users/armand/workspace/storybook/node_modules/babel-runtime/helpers/toConsumableArray.js
 @ /Users/armand/workspace/storybook/node_modules/react-inspector/lib/table-inspector/getHeaders.js
 @ /Users/armand/workspace/storybook/node_modules/react-inspector/lib/table-inspector/TableInspector.js
 @ /Users/armand/workspace/storybook/node_modules/react-inspector/lib/index.js
 @ ../theming/dist/create.js
 @ ../theming/dist/index.js
 @ dll storybook_ui
```

It's look like some of our dependencies are trying to acces to the `core-js/library/*` import. But this import does not exist anymore in `core-js@3`. I do not understand why these dependencies have not `core-js@2` hoisted in there node modules.

Did someone have an idea? (@ndelangen ?)

**Comment by Spy-Seth** [comment]
I will wait to have fix these issues before rebasing again and resolving the conflict.

**Comment by ndelangen** [comment]
A couple of dependencies we use and bundle in our code use corejs 2, and this is causing us some issues.

**Comment by ndelangen** [comment]
I submitted a PR to react-inspector:
https://github.com/xyc/react-inspector/pull/78

**Comment by Spy-Seth** [comment]
`simplebar` dependency have already upgrade https://github.com/uniibu/simplebar/pull/17 😻 The change is available in `simplebar@4.0.0-alpha.5`

File: addons/storyshots/storyshots-core/src/frameworks/angular/loader.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 import hasDependency from '../hasDependency';
 import configure from '../configure';
 
 function setupAngularJestPreset() {
   // Needed to prevent "Zone.js has detected that ZoneAwarePromise `(window|global).Promise` has been overwritten."
   require.requireActual('core-js');
-  require.requireActual('core-js/modules/es6.promise');
+  require.requireActual('core-js/modules/es.promise');
   // require.requireActual('core-js/es6/reflect');
   // require.requireActual('core-js/es7/reflect');
```
Comment (Spy-Seth) [inline addons/storyshots/storyshots-core/src/frameworks/angular/loader.js]: Should we not clean the now outdated commented code about `core-js` like these two?

**Review by ndelangen — APPROVED** [review]

File: addons/a11y/src/components/Report/__snapshots__/HighlightToggle.test.js.snap [hunk 2]
```diff
@@ -37,8 +38,10 @@ exports[`HighlightToggle component should match snapshot 1`] = `
             "HTML_TAGNAME_TEXT_TRANSFORM": "lowercase",
             "HTML_TAG_COLOR": "rgb(168, 148, 166)",
             "OBJECT_NAME_COLOR": "rgb(136, 19, 145)",
+            "OBJECT_PREVIEW_ARRAY_MAX_PROPERTIES": 10,
+            "OBJECT_PREVIEW_OBJECT_MAX_PROPERTIES": 5,
             "OBJECT_VALUE_BOOLEAN_COLOR": "rgb(28, 0, 207)",
-            "OBJECT_VALUE_FUNCTION_KEYWORD_COLOR": "rgb(170, 13, 145)",
+            "OBJECT_VALUE_FUNCTION_PREFIX_COLOR": "rgb(13, 34, 170)",
```
Comment (benoitdion) [inline addons/a11y/src/components/Report/__snapshots__/HighlightToggle.test.js.snap]: are these changes the result of something else?

Comment (ndelangen) [reply addons/a11y/src/components/Report/__snapshots__/HighlightToggle.test.js.snap]: I don't really know TBH, I just updated the snapshots

**Comment by libetl** [comment]
This has broken the dll in my branch

**Pull request merged**